### PR TITLE
feat: Added MiniPlaceholders support

### LINKED ISSUE #1: Add MiniPlaceholders support
https://github.com/MiniPlaceholders/MiniPlaceholders

### PR DESCRIPTION
**Pull request by 4drian3d**
resolves #1

### CODE REVIEW

File: src/main/java/com/mrivanplays/papisigns/loader/PapiSigns.java [hunk 1]
```diff
@@ -35,9 +40,26 @@ public class PapiSigns extends JavaPlugin {
...
+        .filter(PlaceholderProvider::available)
+        .findFirst()
+        .orElse(null);
+
+    if (placeholderProvider == null) {
+        getLogger().info("No Placeholder Provider plugin found. Disabling plugin.");
+        getServer().getPluginManager().disablePlugin(this);
+        return;
+    } else {
+        getLogger().info("Using "+placeholderProvider+" as PlaceholderProvider");
```
Comment (lokka30) [inline src/main/java/com/mrivanplays/papisigns/loader/PapiSigns.java:60]:
```suggestion
        getLogger().info("Using " + placeholderProvider + " as PlaceholderProvider");
```

File: src/main/java/com/mrivanplays/papisigns/loader/PapiSigns.java [hunk 2]
```diff
@@ -35,9 +40,26 @@ public class PapiSigns extends JavaPlugin {
...
+    this.placeholderProvider = Stream.of(
+        new MiniPlaceholdersProvider(),
+        new PlaceholderAPIProvider()
+    )
+        .filter(PlaceholderProvider::available)
+        .findFirst()
+        .orElse(null);
+
+    if (placeholderProvider == null) {
+        getLogger().info("No Placeholder Provider plugin found. Disabling plugin.");
```
Comment (lokka30) [inline src/main/java/com/mrivanplays/papisigns/loader/PapiSigns.java:56]:
```suggestion
        getLogger().warning("No compatible placeholder provider plugin found; disabling plugin.");
```

**Review by lokka30 — CHANGES_REQUESTED** [review]
Awesome, thank you! I have a few tiny changes to recommend.

**Review by lokka30 — APPROVED** [review]

**Review by MrIvanPlays — CHANGES_REQUESTED** [review]
Seems good overall, however I'd like some sort of priority system.
For example what if a server has both MiniPlaceholders and PlaceholderAPI? Which one is going to be prioritized? 

Let's consider that a server runs both MP and PAPI. In that case, we also need a system to distinguish whether a placeholder is a MP or PAPI one.

Please let me know whether you can implement these stuff. If you can't I'm gonna merge this as-is and implement such systems.

**Comment by 4drian3d** [comment]
> Seems good overall, however I'd like some sort of priority system. For example what if a server has both MiniPlaceholders and PlaceholderAPI? Which one is going to be prioritized?
> 
> Let's consider that a server runs both MP and PAPI. In that case, we also need a system to distinguish whether a placeholder is a MP or PAPI one.
> 
> Please let me know whether you can implement these stuff. If you can't I'm gonna merge this as-is and implement such systems.

Currently MiniPlaceholders integration would be prioritized, if someone wanted to use PlaceholderAPI placeholders they could use the https://github.com/MiniPlaceholders/PlaceholderAPI-Expansion expansion.
This could not be done from PlaceholderAPI, that's why I prioritized the MiniPlaceholders integration, since PlaceholderAPI has string placeholders, while MiniPlaceholders has component placeholders.
Nor could you do a multiparse between multiple placeholder integrations because it would be quite resource intensive as it would serialize and deserialize components multiple times, in an event as heavy as the PlayerMoveEvent

**Review by MrIvanPlays — APPROVED** [review]

**Comment by lokka30** [comment]
Sweet - thanks for your contribution @4drian3d!

**Pull request merged**